caravel/templates/caravel/dashboard.html

```diff
@@ -75,26 +75,28 @@ <h4 class="modal-title" id="myModalLabel">Refresh Interval</h4>
         </h2>
       </div>
       <div class="col-md-3">
-        <div class="btn-group pull-right" role="group" >
-          <button type="button" id="refresh_dash" class="btn btn-default" data-toggle="tooltip" title="Force refresh the whole dashboard">
-            <i class="fa fa-refresh"></i>
-          </button>
-          <button type="button" id="refresh_dash_periodic" class="btn btn-default" data-toggle="modal" data-target="#refresh_modal">
-              <i class="fa fa-clock-o" data-toggle="tooltip" title="Decide how frequent should the dashboard refresh"></i>
-          </button>
-          <button type="button" id="filters" class="btn btn-default" data-toggle="tooltip" title="View the list of active filters">
-            <i class="fa fa-filter"></i>
-          </button>
-          <button type="button" id="css" class="btn btn-default {{ "disabled disabledButton" if not dash_edit_perm }} " data-toggle="modal" data-target="#css_modal">
-            <i class="fa fa-css3" data-toggle="tooltip" title="Edit the dashboard's CSS"></i>
-          </button>
-          <a id="editdash" class="btn btn-default {{ "disabled disabledButton" if not dash_edit_perm }} " href="/dashboardmodelview/edit/{{ dashboard.id }}" title="Edit this dashboard's property" data-toggle="tooltip" >
-              <i class="fa fa-edit"></i>
-          </a>
-          <button type="button" id="savedash" class="btn btn-default {{ "disabled disabledButton" if not dash_save_perm }}" data-toggle="tooltip" title="Save the current positioning and CSS">
-            <i class="fa fa-save"></i>
-          </button>
-        </div>
+        {% if dash_edit_perm %}
+          <div class="btn-group pull-right" role="group" >
```

Comment:
Yeah I thought about going React here but I was confused when thinking about how it would play against all of the non-react of the page. It turned out to be surprisingly complicated, and a better candidate for React than I had thought at first.